frontend/public/components/secrets/create-secret.tsx

```diff
@@ -59,116 +84,258 @@ class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretSt
-    secretType={params.type}
-    titleVerb="Create"
-    isCreate={true}
-  />;
-};
-
-export const EditSecret = ({match: {params}, kind}) => <Firehose resources={[{kind: kind, name: params.name, namespace: params.ns, isList: false, prop: 'obj'}]}>
-  <BindingLoadingWrapper fixedKeys={['kind', 'metadata']} titleVerb="Edit" saveButtonText="Save Changes" />
-</Firehose>;
-
-const generateSecret = () => {
-  // http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
-  const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
-  return s4() + s4() + s4() + s4();
-};
-
-class WebHookSecretSubform extends React.Component<WebHookSecretSubformProps, WebHookSecretSubformState> {
+class WebHookSecretForm extends React.Component<WebHookSecretFormProps, WebHookSecretFormState> {
   constructor(props) {
     super(props);
-    this.state = {WebHookSecretKey: this.props.stringData.WebHookSecretKey || ''};
+    this.state = {
+      stringData: {WebHookSecretKey: this.props.stringData.WebHookSecretKey || ''}
+    };
     this.changeWebHookSecretkey = this.changeWebHookSecretkey.bind(this);
     this.generateWebHookSecret = this.generateWebHookSecret.bind(this);
   }
   changeWebHookSecretkey(event) {
     this.setState({
-      WebHookSecretKey: event.target.value
+      stringData: { WebHookSecretKey: event.target.value }
     }, () => this.props.onChange(this.state));
   }
   generateWebHookSecret() {
     this.setState({
-      WebHookSecretKey: generateSecret()
+      stringData: { WebHookSecretKey: generateSecret() }
     }, () => this.props.onChange(this.state));
   }
   render () {
     return <div className="form-group">
       <label className="control-label" htmlFor="webhook-secret-key">Webhook Secret Key</label>
       <div className="input-group">
-        <input className="form-control" id="webhook-secret-key" type="text" name="webhookSecretKey" onChange={this.changeWebHookSecretkey} value={this.state.WebHookSecretKey} required/>
+        <input className="form-control"
+          id="webhook-secret-key"
+          type="text"
+          name="webhookSecretKey"
+          onChange={this.changeWebHookSecretkey}
+          value={this.state.stringData.WebHookSecretKey}
+          aria-describedby="webhook-secret-help"
+          required />
         <span className="input-group-btn">
           <button type="button" onClick={this.generateWebHookSecret} className="btn btn-default">Generate</button>
         </span>
       </div>
-      <p className="help-block">Value of the secret will be supplied when invoking the webhook. </p>
+      <p className="help-block" id="webhook-secret-help">Value of the secret will be supplied when invoking the webhook. </p>
     </div>;
   }
 }
 
+class SourceSecretForm extends React.Component<SourceSecretFormProps, SourceSecretFormState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      authenticationType: this.props.secretType,
```

Comment:
Yeah, I suggest just making this `type`, then it can be handled generically as type by `SecretFormComponent`